src/back-end/lib/resources/feedback.ts

```diff
@@ -0,0 +1,56 @@
+import { getString } from 'back-end/../shared/lib';
+import { AvailableModels, Session, SupportedRequestBodies } from 'back-end/lib/app/types';
+import * as crud from 'back-end/lib/crud';
+import { basicResponse, JsonResponseBody, makeJsonResponseBody, Response } from 'back-end/lib/server';
+import { CreateRequestBody, CreateValidationErrors } from 'shared/lib/resources/feedback';
+import { validateFeedbackText } from 'shared/lib/validators/feedback';
+
+type CreateResponseBody = JsonResponseBody<null | CreateValidationErrors>;
+
+type RequiredModels = 'Feedback';
+
+export type Resource = crud.Resource<SupportedRequestBodies, JsonResponseBody, AvailableModels, RequiredModels, CreateRequestBody, null, Session>;
+
+export const resource: Resource = {
+
+  routeNamespace: 'feedback',
+
+  create(Models) {
+    const FeedbackModel = Models.Feedback;
+    return {
+      async transformRequest(request) {
+        return {
+          rating: getString(request.body.value, 'rating'),
+          text: getString(request.body.value, 'text')
+        }
+      },
+      async respond(request): Promise<Response<CreateResponseBody, Session>> {
+        const respond = (code: number, body: null | CreateValidationErrors) => basicResponse(code, request.session, makeJsonResponseBody(body));
+
+        // Create feedback object
+        const feedback = new FeedbackModel({
+          createdAt: new Date(),
+          text: request.body.text,
+          rating: request.body.rating
```

Comment:
The compiler should be showing an error here, as `rating` is a string, not one of `good`, `meh` or `bad`.

Reply:
The `rating` needs to be validated like `feedbackText`